project-geometry/src/main/java/com/esri/arcgismaps/sample/projectgeometry/MainActivity.kt

```diff
@@ -0,0 +1,158 @@
+        activityMainBinding.infoTextView
+    }
+
+    // setup the red pin marker image as bitmap drawable
+    private val markerDrawable: BitmapDrawable by lazy {
+        // load the bitmap from resources and create a drawable
+        val bitmap = BitmapFactory.decodeResource(resources, R.drawable.pin_symbol)
+        BitmapDrawable(resources, bitmap)
+    }
+
+    // setup the red pin marker as a Graphic
+    private val markerGraphic: Graphic by lazy {
+        // creates a symbol from the marker drawable
+        Graphic().apply {
+            symbol = PictureMarkerSymbol(markerDrawable).apply {
+                // resize the symbol into a smaller size
+                width = 30f
+                height = 30f
+                // offset in +y axis so the marker spawned
+                // is right on the touch point
+                offsetY = 25f
+            }
+        }
+    }
+
+    // creates a graphic overlay
+    private val graphicsOverlay: GraphicsOverlay = GraphicsOverlay()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        // authentication with an API key or named user is
+        // required to access basemaps and other location services
+        ArcGISEnvironment.apiKey = ApiKey.create(BuildConfig.API_KEY)
+        lifecycle.addObserver(mapView)
+
+        // configure mapView assignments
+        mapView.apply {
+            // create and add a map with a navigation night basemap style
+            map = ArcGISMap(BasemapStyle.ArcGISNavigationNight)
+            // add the graphics overlay to display marker graphics
+            graphicsOverlays.add(graphicsOverlay)
+            // set the default viewpoint to Redlands,CA
+            setViewpoint(Viewpoint(34.058, -117.195, 5e4))
+        }
+
+        lifecycleScope.launch {
+            // check if the map has loaded successfully
+            mapView.map?.load()?.onSuccess {
+                // capture and collect when the user taps on the screen
+                mapView.onSingleTapConfirmed.collect { event ->
+                    event.mapPoint?.let { point -> geoViewTapped(point) }
+                }
+            }?.onFailure {
+                // if map load failed, show the error
+                showError("Error Loading Map", mapView)
+            }
+        }
+    }
+
+    /**
+     * Handles the SingleTapEvent by drawing a marker and performs a Spatial reference transformation
+     * of the tapped Location using GeometryEngine and displays the result
+     */
+    private fun geoViewTapped(point: Point) {
+        // show the marker where the user tapped on the map
+        graphicsOverlay.graphics.apply {
+            // this removes previously added markers
+            clear()
+            // add a new marker graphic
+            add(markerGraphic.apply {
+                // updates the geometry location to the current tap
+                geometry = point
+            })
+        }
```

Comment:
The nesting here makes this a bit harder to read than it needs to be. I think first moving the marker graphic and then updating the graphics overlay would be clearer
```suggestion
        markerGraphic.geometry = point
        graphicsOverlay.graphics.apply {
            // this removes previously added markers
            clear()
            // add a new marker graphic
            add(markerGraphic)
        }
```
Although technically there's no need to clear and re-add the marker graphic if it's already there. Its position should update automatically when its geometry is changed.